src/event_loop/src/lib.rs

```diff
@@ -14,44 +14,106 @@ use std::cmp;
 use window::Window;
 use input::{ AfterRenderArgs, Event, IdleArgs, RenderArgs, UpdateArgs };
 
-/// A trait for create event iterator from window.
+/// A trait for creating an event iterator for a window.
+/// 
+/// This is the current ideomatic way of handling an event loop in Piston:
+/// ```
+/// let mut events: WindowEvents = window.events();
+/// while let Some(event) = events.next() {
```

Comment:
`events.next(&mut window)`

You need to pass in the window since the event loop polls events from it.